crypto/helpers.go

```diff
@@ -152,7 +233,91 @@ func RSAEncrypt(plaintext []byte, publicKey *rsa.PublicKey) ([]byte, error) {
 	}
 }
 
-func RSADecrypt(ciphertext []byte, privateKey *rsa.PrivateKey) ([]byte, error) {
+func eciesEncrypt(plaintext []byte, publicKey *ecdsa.PublicKey) ([]byte, error) {
```

Comment:
Initially I had two separate functions RSADecrypt() and ECIESDecrypt() that  were required to be explicitly used. I thought it would be better to have a general Decrypt() function that was smart enough to decrypt either type. the private eciesDecrypt() and rsaDecrypt() functions are now used for code clarity and not intended to be used outside of the package. If decryption is required, one should use the generic Decrypt() function.